api/build.gradle

```diff
@@ -3,13 +3,83 @@ import io.swagger.codegen.DefaultGenerator
 import org.pmiops.workbench.tooling.GenerateAPIListingTask
 import org.pmiops.workbench.tooling.IncrementalHotSwapTask
 
+// Runs before all tasks. Sets up properties and dependencies for the build
+// process itself.
+buildscript {
+  // Project-wide constants.
+  ext {
+    ELASTICSEARCH_VERSION = '6.8.3'
+    GAE_VERSION = '1.9.64'
+    GSON_VERSION = '2.8.5'
+    HIBERNATE_VERSION = '5.2.10.Final'
+    JACKSON_DATABIND_VERSION = '2.9.10.1'
+    JACKSON_VERSION = '2.9.10'
+    JODA_VERSION = '2.10'
+    KOTLIN_VERSION = '1.3.50'
+    MAPSTRUCT_VERSION = '1.3.1.Final'
+    MOCKITO_KOTLIN_VERSION = '2.2.0'
+    OKHTTP_VERSION = '2.7.5'
+    OPENCENSUS_VERSION = '0.23.0'
+    SPRINGFOX_VERSION = '2.6.1'
+    SPRING_BOOT_VERSION = '1.5.22.RELEASE'
+    SWAGGER_CODEGEN_VERSION = '2.2.3'
+  }
+
+  repositories {
+    // Bintray's repository - a fast Maven Central mirror & more
+    jcenter()
+    maven {
+      url "https://plugins.gradle.org/m2/"
+    }
+  }
+
+  // Buildscript-specific dependencies (uses buildscript-scoped repositories). This
+  // closure ensures allows the buildscript dependencies (e.g. for plgins) to be managed
+  // independently from the main project's application code's dependencies
+  dependencies {
+    classpath 'com.diffplug.spotless:spotless-plugin-gradle:3.23.0'
+    classpath 'com.google.cloud.tools:appengine-gradle-plugin:1.3.5'
+    classpath 'gradle.plugin.org.hidetake:gradle-swagger-generator-plugin:2.12.0'
+    // Note: if you plan to upgrade the version of swagger-codegen beyond 2.2.3, be aware of two
+    // implicit dependencies: (1) the set of "generatedCompile" Gradle dependencies need to be updated
+    // to match the modules imported by the Swagger codegen templates, and (2) some template files
+    // are overridden with forked templates within this repo; they will likely need to be re-copied from
+    // the newer release version and have AoU-specific modifications re-applied.
+    classpath "io.swagger:swagger-codegen:$project.ext.SWAGGER_CODEGEN_VERSION"
+    classpath 'net.ltgt.gradle:gradle-apt-plugin:0.21'
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$project.ext.KOTLIN_VERSION"
+    classpath "org.jetbrains.kotlin:kotlin-noarg:$project.ext.KOTLIN_VERSION"
+    classpath 'org.owasp:dependency-check-gradle:5.1.0'
+    classpath "org.springframework.boot:spring-boot-gradle-plugin:$project.ext.SPRING_BOOT_VERSION"
+  }
+}
+
+project.ext['hibernate.version'] = project.ext.HIBERNATE_VERSION
```

Comment:
How does this relate to buildscript.ext.HIBERNATE_VERSION? If you have a clear idea of what this line is doing / why it exists, a short comment here might help.